libs/entitlement/tools/securitymanager-scanner/src/main/java/org/elasticsearch/entitlement/tools/securitymanager/scanner/Main.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.entitlement.tools.securitymanager.scanner;
+
+import org.elasticsearch.core.SuppressForbidden;
+import org.elasticsearch.entitlement.tools.Utils;
+import org.objectweb.asm.ClassReader;
+
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Set;
+
+public class Main {
+
+    static final Set<String> excludedModules = Set.of("java.desktop");
+
+    private static void identifySMChecksEntryPoints() throws IOException {
+
+        FileSystem fs = FileSystems.getFileSystem(URI.create("jrt:/"));
+
+        var moduleExports = Utils.findModuleExports(fs);
+
+        var callers = new HashMap<String, List<SecurityCheckClassVisitor.CallerInfo>>();
+        var visitor = new SecurityCheckClassVisitor(callers);
+
+        try (var stream = Files.walk(fs.getPath("modules"))) {
+            stream.filter(x -> x.toString().endsWith(".class")).forEach(x -> {
```

Comment:
Given that `x` is used in multiple places here, it might make the code easier to follow if it had a descriptive name.